File: java/behavioral_pattern/observer/StaticsDisplay.java
package observer;

import java.util.ArrayList;
import java.util.List;

public class StaticsDisplay implements Observer {
    private List<Float> temperatureHistory = new ArrayList<>();

    @Override
    public void update(float temperature, float humidity, float pressure) {
        temperatureHistory.add(temperature);
        display();
    }

    public void display() {
        float sum = 0;
        for (float temp : temperatureHistory) {
            sum += temp;
        }
        float avg = sum / temperatureHistory.size();
        System.out.println("Avg/Max/Min temperature: " + avg + "/" + getMax() + "/" + getMin());
    }

    private float getMax() {
        float max = Float.MIN_VALUE;
        for (float temp : temperatureHistory) {
            if (temp > max) max = temp;
        }

        return max;
    }

    private float getMin() {
        float min = Float.MAX_VALUE;
        for (float temp : temperatureHistory) {
            if (temp < min) min = temp;
        }
        return min;
    }
}

File: typescript/behavioral_pattern/src/observer/StaticsDisplay.ts
import { Observer } from "./Observer";

class StaticDisplay implements Observer {
    private temperatureHistory: number[] = [];

    public update(temperature: number, humidity: number, pressure: number): void {
        this.temperatureHistory.push(temperature);
        this.display();
    }

    public display(): void {
        let sum = 0;
        for (const temp of this.temperatureHistory) {
            sum += temp;
        }
        const avg = sum / this.temperatureHistory.length;
        console.log("Avg/Max/Min temperature: " + avg + "/" + this.getMax() + "/" + this.getMin());
    }

    private getMax(): number {
        let max = Number.MIN_VALUE;
        for (const temp of this.temperatureHistory) {
            if (temp > max) max = temp;
        }

        return max;
    }

    private getMin(): number {
        let min = Number.MAX_VALUE;
        for (const temp of this.temperatureHistory) {
            if (temp < min) min = temp;
        }

        return min;
    }

}

export default StaticDisplay;